src/core/data-abbr.js

```diff
@@ -0,0 +1,43 @@
+// @ts-check
+// Module core/data-abbr
+// - Finds all elements with data-abbr attribute and processes them.
+import { showInlineWarning } from "./utils";
+export const name = "core/dfn-abbr";
+
+export function run() {
+  document.querySelectorAll("[data-abbr]").forEach(elem => {
+    const { localName } = elem;
+    switch (localName) {
+      case "dfn":
+        // @ts-ignore
+        processDfnElement(elem);
+        break;
+      default: {
+        const msg =
+          `[\`data-dfn\`](https://github.com/w3c/respec/wiki/data-abbr)` +
+          ` attribute not supported on \`${localName}\` elements.`;
+        showInlineWarning(elem, msg, "Error: unsupported.");
+      }
+    }
+  });
+}
+/**
+ * @param {HTMLElement} dfn
+ */
+function processDfnElement(dfn) {
+  if (dfn.dataset.abbr === "") {
+    // Generates abbreviation from textContent
+    // e.g., "Permanent Account Number" -> "PAN"
+    dfn.dataset.abbr = dfn.textContent
+      .match(/\b([a-z])/gi)
+      .join("")
+      .toUpperCase();
+  }
+  const fullForm = dfn.textContent.trim();
+  const { abbr } = dfn.dataset;
+  dfn.insertAdjacentHTML(
+    "afterend",
+    ` (<abbr title="${fullForm}">${abbr}</abbr>)`
+  );
+  dfn.dataset.lt = dfn.dataset.lt ? `${dfn.dataset.lt}|${abbr}` : abbr;
```

Comment:
This means we cannot link to the definition using ```<a class="dfnlink">foo bar (FB)</a>```.
Although not a huge deal, I think it might be counter intuitive as visually the definition after rendering reads foo bar (FB). The user may want to refer it as such.

Reply:
Understood. However, I've never seen that kind linking in a spec in practice. It's always:

 * `<a>FB</a>` or `<a>foo bar</a>`, but never `<a class="dfnlink">foo bar (FB)</a>`.

Reply:
Alright. I meant ```<a>foo bar (FB)</a>``` and not the class too. 
Entirely your call of course.

Reply:
yeah, still the same. We can always add it easily if it comes up in practice or someone asks for it.